src/VCR/Configuration.php

```diff
@@ -14,6 +14,11 @@
  */
 class Configuration
 {
+    public function __construct() {
+        $this->matchObserver = new MatchObserver();
```

Comment:
The configuration is currently a simple container for configuration data. All instance creation should happen in the factory and then handed in where it is needed, see "inversion of control".

When instance creation logic moves to the `VCRFactory` the `matcherObserver` should not be needed in the configuration.